fastparquet/writer.py

```diff
@@ -657,71 +692,89 @@ def write(filename, data, row_group_offsets=50000000, encoding="PLAIN",
         to fixed-length strings of the given length for the given columns
         before writing, potentially providing a large speed
         boost.
+    append: bool (False)
+        If False, construct data-set from scratch; if True, add new row-group(s)
+        to existing data-set. In the latter case, the data-set must exist,
+        and the schema must match the input data.
 
     Examples
     --------
     >>> fastparquet.write('myfile.parquet', df)  # doctest: +SKIP
     """
     sep = sep_from_open(open_with)
-    if file_scheme != 'simple' or isinstance(data, pd.core.groupby.DataFrameGroupBy):
-        mkdirs(filename)
-        fn = sep.join([filename, '_metadata'])
-    else:
-        fn = filename
     if isinstance(row_group_offsets, int):
         l = len(data)
         nparts = (l - 1) // row_group_offsets + 1
         chunksize = (l - 1) // nparts + 1
         row_group_offsets = list(range(0, l, chunksize))
-    with open_with(fn) as f:
-        f.write(MARKER)
-
-        if write_index or write_index is None and not (
-                isinstance(data.index, pd.RangeIndex) and
-                data.index._start == 0 and
-                data.index._stop == len(data.index) and
-                data.index._step == 1 and data.index.name is None):
-            data = data.reset_index()
-        ignore = partition_on if file_scheme != 'simple' else []
-        fmd = make_metadata(data, has_nulls=has_nulls, ignore_columns=ignore,
-                            fixed_text=fixed_text)
+    if write_index or write_index is None and not (
+            isinstance(data.index, pd.RangeIndex) and
+            data.index._start == 0 and
+            data.index._stop == len(data.index) and
+            data.index._step == 1 and data.index.name is None):
```

Comment:
I think that this is trying to determine if the index is naive like `0..n`.  I'm usually in favor of nesting one-off logic into small functions for fear of making the code too indirect.  This seems like a good candidate for a small utils function though.